File: rust/src/questions/k01.rs
use crate::shared::traits::executable::Executable;

pub struct K01 {}

impl Executable for K01
{
    fn question_1(&self) -> anyhow::Result<()>
    {
        println!("Hello World!　ようこそRust言語の世界へ！");
        Ok(())
    }

    fn question_2(&self) -> anyhow::Result<()>
    {
        println!("Hello World!");
        println!("ようこそ");
        println!("Rust言語の世界へ！");
        Ok(())
    }

    fn question_3(&self) -> anyhow::Result<()>
    {
        println!("整数：{}", 12345);
        println!("実数：{}", 123.456789);
        println!("文字：{}", 'X');
        println!("文字列：{}", "ABCdef");
        Ok(())
    }

    fn question_4(&self) -> anyhow::Result<()>
    {
        println!("              ##");
        println!("             #  #");
        println!("             #  #");
        println!("            #    #");
        println!("           #      #");
        println!("         ##        ##");
        println!("       ##            ##");
        println!("    ###                ###");
        println!(" ###       ##    ##       ###");
        println!("##        #  #  #  #        ##");
        println!("##         ##    ##         ##");
        println!(" ##     #            #     ##");
        println!("  ###     ##########     ###");
        println!("     ###              ###");
        println!("        ##############");
        Ok(())
    }
}

File: TypeScript/assignments/K01.ts
import Executable from "../shared/Executable.ts";

export default class K01 extends Executable {
    question1(): void {
        console.log("Hello World! ようこそTypeScript言語の世界へ！");
    }

    question2(): void {
        console.log("Hello World!");
        console.log("ようこそ");
        console.log("TypeScript言語の世界へ！");
    }

    question3(): void {
        console.log("整数：" + 12345);
        console.log("実数：" + 123.456789);
        console.log("文字：" + 'A');
        console.log("文字列：" + "ABCdef");
    }

    question4(): void {
        console.log("              ##");
        console.log("             #  #");
        console.log("             #  #");
        console.log("            #    #");
        console.log("           #      #");
        console.log("         ##        ##");
        console.log("       ##            ##");
        console.log("    ###                ###");
        console.log(" ###       ##    ##       ###");
        console.log("##        #  #  #  #        ##");
        console.log("##         ##    ##         ##");
        console.log(" ##     #            #     ##");
        console.log("  ###     ##########     ###");
        console.log("     ###              ###");
        console.log("        ##############");
    }
}